databricks/provider.go

```diff
@@ -209,6 +265,11 @@ func providerConfigure(d *schema.ResourceData, providerVersion string) (interfac
 		if token, ok := d.GetOk("token"); ok {
 			config.Token = token.(string)
 		}
+		if config.Host == "" || config.Token == "" {
+			if err := tryDatabricksCliConfigFile(d, &config); err != nil {
+				return nil, err
```

Comment:
Since error does not contain stack trace as java/c# exception - it is always a good idea to wrap error into another error as you go up the stack for better debugability. You can use [errors.Wrapf](https://godoc.org/github.com/pkg/errors#Wrapf) for that.
Here it would be `return nil, errors.Wrap(err, "getting databricks credentials from config failed")`